flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -606,7 +608,12 @@ private void releaseEmptyTaskManager(ResourceID resourceId) {
 
 		final TaskManagerLocation taskManagerLocation;
 		try {
-			taskManagerLocation = TaskManagerLocation.fromUnresolvedLocation(unresolvedTaskManagerLocation);
+			final boolean retrieveTaskManagerHostName = this.jobMasterConfiguration.getConfiguration()
+					.getBoolean(JobManagerOptions.RETRIEVE_TASK_MANAGER_HOSTNAME);
```

Comment:
I was wondering whether it makes sense to move this local variable to a field, and do this lookup in the constructor only once?

Reply:
Thank you for the suggestion, I have moved this variable to a field in *JobMaster*.